tests/tests/connectors/test-class-connector-users.php

```diff
@@ -0,0 +1,307 @@
+			->method( 'log' )
+			->withConsecutive(
+				array(
+					$this->equalTo(
+						_x(
+							'%1$s\'s account was deleted (%2$s)',
+							'1: User display name, 2: User roles',
+							'stream'
+						)
+					),
+					$this->equalTo(
+						array(
+							'display_name' => 'TestGuy',
+							'roles'        => 'Subscriber',
+						)
+					),
+					$this->equalTo( $user_id ),
+					$this->equalTo( 'users' ),
+					$this->equalTo( 'deleted' ),
+					$this->equalTo( 0 )
+				),
+				array(
+					$this->equalTo( esc_html__( 'User account #%d was deleted', 'stream' ) ),
+					$this->equalTo(
+						array(
+							'display_name' => $user_id,
+							'roles'        => '',
+						)
+					),
+					$this->equalTo( $user_id ),
+					$this->equalTo( 'users' ),
+					$this->equalTo( 'deleted' ),
+					$this->equalTo( 0 )
+				)
+			);
+
+		// Do stuff.
+		wp_delete_user( $user_id );
+		do_action( 'deleted_user', $user_id );
+
+		// Check callback test action.
+		$this->assertFalse( 0 === did_action( 'wp_stream_test_callback_delete_user' ) );
+		$this->assertFalse( 0 === did_action( 'wp_stream_test_callback_deleted_user' ) );
+	}
+
+	public function test_callback_set_user_role() {
+		// Create user.
+		$user_id = self::factory()->user->create( array( 'display_name' => 'TestGuy' ));
+		$user    = get_user_by( 'id', $user_id );
+
+		// Expected log calls.
+		$this->mock->expects( $this->once() )
+			->method( 'log' )
+			->with(
+				$this->equalTo(
+					_x(
+						'%1$s\'s role was changed from %2$s to %3$s',
+						'1: User display name, 2: Old role, 3: New role',
+						'stream'
+					)
+				),
+				$this->equalTo(
+					array(
+						'display_name' => 'TestGuy',
+						'old_role'     => 'Subscriber',
+						'new_role'     => 'Editor',
+					)
+				),
+				$this->equalTo( $user_id ),
+				$this->equalTo( 'profiles' ),
+				$this->equalTo( 'updated' )
+			);
+
+		// Do stuff.
+		$user->set_role( 'editor' );
+
+		// Check callback test action.
+		$this->assertFalse( 0 === did_action( 'wp_stream_test_callback_set_user_role' ) );
+	}
+
+	public function test_callback_set_current_user() {
+		// Create user.
+		$user_id = self::factory()->user->create( array( 'display_name' => 'TestGuy' ));
+
+		// Expected log calls.
+		$this->mock->expects( $this->exactly( 2 ) )
+			->method( 'log' )
+			->withConsecutive(
+				array(
+					$this->equalTo( __( '%s logged in', 'stream' ) ),
+					$this->equalTo( array( 'display_name' => 'TestGuy' ) ),
+					$this->equalTo( $user_id ),
+					$this->equalTo( 'sessions' ),
+					$this->equalTo( 'login' ),
+					$this->equalTo( $user_id )
+				),
+				array(
+					$this->equalTo( __( '%s logged in', 'stream' ) ),
+					$this->equalTo( array( 'display_name' => 'TestGuy' ) ),
+					$this->equalTo( $user_id ),
+					$this->equalTo( 'sessions' ),
+					$this->equalTo( 'login' ),
+					$this->equalTo( $user_id )
+				)
+			);
+
+		// Do stuff.
+		wp_set_current_user( $user_id );
+		wp_set_current_user( 0 );
```

Comment:
@kidunot89 If the previous call generates an entry in the log, then ideally calling the same function with `0` as an argument generates an entry as well, otherwise we'll end up with two subsequent "logged in" events and no "log out". See previous comments, please.

Reply:
Not necessary, Stream is designed to target adminstrative actions primarily